eth/client.go

```diff
@@ -155,6 +165,62 @@ func (cl *Client) SubscribeToBlocks() (*BlockSubscription, error) {
 	return &BlockSubscription{sub, blocks, quit}, nil
 }
 
+// SubscribeToBlocksStartingFrom subscribes the client to the mined Ethereum
+// blocks starting from the given block number.
+func (cl *Client) SubscribeToBlocksStartingFrom(startBlockNumber *big.Int) (*BlockSubscription2, error) {
+	headers := make(chan *types.Header)
+	blocks := make(chan *tee.Block)
+	errChan := make(chan error)
+
+	ctx, cancel := NewDefaultContext()
+	defer cancel()
+
+	headerSub, err := cl.SubscribeNewHead(ctx, headers)
+	if err != nil {
+		return nil, fmt.Errorf("subscribing to blockchain head: %w", err)
+	}
+
+	blockSub := &BlockSubscription2{headerSub, blocks, errChan, startBlockNumber}
+
+	run := func() error {
+		for {
+			select {
+			case err := <-headerSub.Err():
+				if err != nil {
+					errChan <- fmt.Errorf("subscription error: %w", err)
```

Comment:
you write into the err chan here and then return the error, which will write it again?

Reply:
nice catch